### PR TITLE
Detect ranges correctly when there are multiple separators

### PR DESCRIPTION
**Pull request by chiragjn**
Detect start date and end date correctly in cases like

'I want to book travel from 13.05.19 to 15.06.19'
'I want to book travel from 13.05.19 to 15.06.19 and 17.06.19 to 18.07.19'
'I want to book travel from 13.05.19 to 15.06.19 to 17.06.19 to 18.07.19'

The approach is to first split by 'and' and 'or', then process each part by finding pairs and avoiding overlaps

### CODE REVIEW

File: ner_v2/detectors/temporal/date/date_detection.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # coding=utf-8
+import itertools
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/date_detection.py]: W1618 [34mimport missing `from __future__ import absolute_import`[0m
F401 'itertools' imported but unused

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by chiragjn** [comment]
retest this please

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Pull request merged**